Social/Facebook/Oauth.php

```diff
@@ -0,0 +1,189 @@
+ */
+
+namespace Integrated\Bundle\SocialBundle\Social\Facebook;
+
+use Facebook\Exceptions\FacebookResponseException;
+use Facebook\Exceptions\FacebookSDKException;
+use Facebook\Facebook;
+use Integrated\Bundle\SocialBundle\Oauth\OauthInterface;
+use Integrated\Common\Channel\Connector\Config\OptionsInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * Class Oauth
+ * @package Integrated\Bundle\SocialBundle\Social\Facebook
+ */
+class Oauth implements OauthInterface
+{
+    private $facebook_app_id;
+
+    private $facebook_app_id_secret;
+
+    /**
+     * @var RequestStack
+     */
+    private $requestStack;
+
+    /**
+     * Oauth constructor.
+     * @param $facebook_app_id
+     * @param $facebook_app_id_secret
+     * @param RequestStack $requestStack
+     */
+    public function __construct($facebook_app_id, $facebook_app_id_secret, RequestStack $requestStack)
+    {
+        $this->facebook_app_id = $facebook_app_id;
+        $this->facebook_app_id_secret = $facebook_app_id_secret;
+        $this->requestStack = $requestStack->getCurrentRequest();
+    }
+
+    /**
+     * @param $connector
+     * @param $admin_url
+     * @return string
+     */
+    public function login($connector, $admin_url)
+    {
+        $fb = new Facebook([
+            'facebook_app_id' => $this->facebook_app_id,
+            'app_secret' => $this->facebook_app_id_secret,
+            'default_graph_version' => 'v2.2'
+        ]);
+
+        $helper = $fb->getRedirectLoginHelper();
+
+        $this->requestStack = $this->requestStack->createFromGlobals();
+
+        $loginUrl = $helper->getLoginUrl(
+            $this->requestStack->server->get('REQUEST_SCHEME')
+            . "://"
+            . $this->requestStack->server->get('HTTP_HOST')
+            . "/"
+            . $admin_url
+            . '/connector/config/'
+            . rawurlencode($connector),
+            ['publish_actions']
+        );
+
+        return $loginUrl;
+    }
+
+    /**
+     * @param OptionsInterface $options
+     * @return array
+     * @throws FacebookSDKException
+     * @throws \Exception
+     */
+    public function callback(OptionsInterface $options)
+    {
+        $fb = new Facebook([
+            'facebook_app_id' => $this->facebook_app_id,
+            'app_secret' => $this->facebook_app_id_secret,
+            'default_graph_version' => 'v2.2',
+        ]);
+
+        $helper = $fb->getRedirectLoginHelper();
+
+        try {
+            $accessToken = $helper->getAccessToken();
+        } catch (FacebookResponseException $e) {
+            // When Graph returns an error
+            throw new \Exception('Graph returned an error: ' . $e->getMessage());
+        } catch (FacebookSDKException $e) {
+            // When validation fails or other local issues
+            throw new \Exception('Facebook SDK returned an error: ' . $e->getMessage());
+        }
+
+        if (!isset($accessToken)) {
```

Comment:
It's not possible for this variable to not be set so there is no reason for this isset check